### PR TITLE
First view to Top Movies app

### PR DESCRIPTION
**Pull request by olegdaniv**
I have written this app to learn and remind  some features like:
- сustom view
- recycleView
- viewModel

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
     package="com.example.topmovies">
-
+<uses-permission android:name="android.permission.INTERNET"/>
```
Comment (MonstraL) [inline app/src/main/AndroidManifest.xml]: Please format the code. There must be the tab space here

File: app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
+import com.example.topmovies.databinding.ItemLayoutBinding
+import com.example.topmovies.model.Movie
+import com.example.topmovies.unit.IMAGE_SIZE
+import com.example.topmovies.unit.REPLACE_AFTER
+
+class MovieAdapter : RecyclerView.Adapter<MovieAdapter.MainViewHolder>() {
+    var movies = mutableListOf<Movie>()
```
Comment (MonstraL) [inline app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt]: It is better to replace with an emptyList

File: app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    fun setMovieList(movies: List<Movie>) {
+        this.movies = movies.toMutableList()
```
Comment (MonstraL) [inline app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt]: Why do you need a MutableList here?

**Comment by MonstraL** [comment]
https://kotlinlang.org/docs/coding-conventions.html
@OlegDaniv Please read these rules and edit your code

File: app/src/main/java/com/example/topmovies/view/CircleAvatarView.kt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    private var text: String = ""
+    private lateinit var imageBitmap: Bitmap
+    private var diameter: Int = 0
+    private val destiny = context.resources.displayMetrics.density
+
+    init {
+        setLayerType(LAYER_TYPE_SOFTWARE, null)
+
+        if (attributeSet != null) {
+            val ta =
```
Comment (MonstraL) [inline app/src/main/java/com/example/topmovies/view/CircleAvatarView.kt]: "The names should make it clear what the purpose of the entity is"

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (olegdaniv) [reply app/src/main/AndroidManifest.xml]: ok, I forgot to do com + alt + L to my manifest file

File: app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt (hunk 2 above)
Comment (olegdaniv) [reply app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt]: OK, but I don't understand why it's so important?

Comment (MonstraL) [reply app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt]: I see no reason to make a mutableList here. It's better not to give the ability to change lists where it's not necessary

File: app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt (hunk 3 above)
Comment (olegdaniv) [reply app/src/main/java/com/example/topmovies/adapter/MovieAdapter.kt]: Sure, in this case, I don't need MutableList.

File: app/src/main/java/com/example/topmovies/view/CircleAvatarView.kt (hunk 4 above)
Comment (olegdaniv) [reply app/src/main/java/com/example/topmovies/view/CircleAvatarView.kt]: I'll consider that.

File: app/src/main/java/com/example/topmovies/retrofit/ApiServer.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.GET
+
+interface ApiServer {
+    @GET(TOP_100_MOVIES)
+    fun getAllMovies(): Call<MovieObject>
+
+    companion object {
+        var ApiService: ApiServer? = null
```
Comment (MonstraL) [inline app/src/main/java/com/example/topmovies/retrofit/ApiServer.kt]: This is a violation of naming conventions

Comment (olegdaniv) [reply app/src/main/java/com/example/topmovies/retrofit/ApiServer.kt]: Alright, what do you think is a good name for this class?

Comment (MonstraL) [reply app/src/main/java/com/example/topmovies/retrofit/ApiServer.kt]: The problem is not the class name. Variable name must start with a lowercase

Comment (olegdaniv) [reply app/src/main/java/com/example/topmovies/retrofit/ApiServer.kt]: Yeah, sorry, I missed that one

File: app/src/main/java/com/example/topmovies/unit/Constants.kt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package com.example.topmovies.unit
+
+const val BASE_URL = "https://imdb-api.com/"
+const val TOP_100_MOVIES = "en/API/MostPopularMovies/k_wfrg16mx"
+const val REPLACE_AFTER = "._V1_"
+const val IMAGE_SIZE = "UX126_CR6,0,126,126_AL_.jpg"
```
Comment (YuriiTsap) [inline app/src/main/java/com/example/topmovies/unit/Constants.kt]: new line

File: app/src/main/res/layout/item_layout.xml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    android:layout_height="wrap_content"
+    android:layout_marginStart="@dimen/layout_margin"
+    android:layout_marginTop="@dimen/layout_margin"
+    android:layout_marginEnd="@dimen/layout_margin"
+    app:cardBackgroundColor="@color/mint_macaron"
+    app:cardCornerRadius="@dimen/item_cardView_corner_radius"
+    app:cardElevation="5dp"
+    app:cardMaxElevation="5dp"
+    app:contentPadding="10dp"
+    app:contentPaddingBottom="10dp">
```
Comment (YuriiTsap) [inline app/src/main/res/layout/item_layout.xml]: - extract to values
- should be dividable by 2

**Pull request merged**